### PR TITLE
Optional options for Blame.file

### LINKED ISSUE #808: Options should be optional in Blame.file
According to the libgit2 docs if the `options` aren't passed in default to a newly initialized `BlameOptions`

### PR DESCRIPTION
**Pull request by smith-kyle**
This should fix #808

### CODE REVIEW

File: lib/blame.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Retrieve the blame of a file
+ *
+ * @param {Repository} repo that contains the file
+ * @param {String} path to the file to get the blame of
+ * @param {BlameOptions} [options] Options for the blame
+ */
+Blame.file = function(repo, path, options) {
+  if (!options) {
```
Comment (johnhaley81) [inline lib/blame.js]: Do https://github.com/nodegit/nodegit/blob/master/lib/checkout.js#L18 instead please.

**Comment by smith-kyle** [comment]
It looks like `normalizeOptions` will return null if `!options` is `true` but I'm trying to instantiate options if they aren't passed in. Maybe something like:
```javascript
options = normalizeOptions(options)
if (options === null) {
  options = new BlameOptions();
}
```

**Comment by johnhaley81** [comment]
libgit2 should handle the null case. If it's null libgit2 creates the default. We shouldn't need to pass it in. At least we don't in other areas. Is this different?

**Comment by smith-kyle** [comment]
Thanks for that. I don't know why I didn't think of libgit2 taking care of the default :+1:

**Comment by johnhaley81** [comment]
np :+1:

**Pull request merged**